### PR TITLE
fix(config): Avoid alias `in_flight_limit`

### LINKED ISSUE #5920: `request.in_flight_limit` alias no longer works
This appears to be a regression from 0.11.1 as it works there; we should fix before 0.12 cc/ @jamtur01 @binarylogic 

### Vector Version
```
vector 0.12.0 (gb448cf4 x86_64-unknown-linux-gnu 2021-01-07)
```

### Vector Configuration File

```toml
[sources.in]
type = "stdin"

[sinks.company_unfurler_http]
type = "http"
inputs = ["in"]
uri = "http://example.com"
encoding.codec = "json"

[sinks.company_unfurler_http.request]
in_flight_limit = 100
rate_limit_duration_secs = 5
rate_limit_num = 60
```

### Debug Output
```
➜  vector git:(master) ./target/debug/vector --config /tmp/http.toml -vvv 
Jan 07 19:51:00.598  INFO vector::app: Log level is enabled. level="vector=trace,codec=trace,file_source=trace,tower_limit=trace,rdkafka=trace"
Jan 07 19:51:00.598  INFO vector::sources::host_metrics: PROCFS_ROOT is unset. Using default '/proc' for procfs root.
Jan 07 19:51:00.598  INFO vector::sources::host_metrics: SYSFS_ROOT is unset. Using default '/sys' for sysfs root.
Jan 07 19:51:00.598  INFO vector::app: Loading configs. path=[("/tmp/http.toml", None)]
Jan 07 19:51:00.612 ERROR vector::cli: Configuration error. error=unknown field `in_flight_limit` for key `sinks.company_unfurler_http` at line 10 column 1
```

### Expected Behavior

Vector sets the concurrency limit to 100.

### Actual Behavior

Vector fails to start.

### PR DESCRIPTION
**Pull request by ktff**
Closes #5920 

A case of serde-rs/serde#1504.

### CODE REVIEW

File: src/sinks/util/service.rs [hunk 1]
```diff
@@ -206,6 +212,17 @@ impl<T: ConcurrencyOption> TowerRequestConfig<T> {
...
+
+    fn concurrency(&self) -> &T {
+        match (self.concurrency.is_none(), self.in_flight_limit.is_none()) {
+            (_, true) => &self.concurrency,
+            (true, false) => &self.in_flight_limit,
+            (false, false) => {
+                warn!("`in_flight_limit` option has been renamed to `concurrency`. Ignoring `in_flight_limit` and using `concurrency` option");
+                &self.concurrency
+            }
+        }
```
Comment (bruceg) [inline src/sinks/util/service.rs:224]: I'd prefer this logic be rewritten in terms of either `is_some` or the actual values (ie `match (&self.concurrency, &self.in_flight_limit)`) I had to read this a couple times before the `true`/`false` values made sense.

**Review by bruceg — COMMENTED** [review]
Thanks, looks good except for my beef with the `.is_none()` logic. Given the problems we've had with this, I think a test would be in order as well.

File: src/sinks/util/service.rs [hunk 2]
```diff
@@ -129,6 +129,9 @@ impl<'de> Deserialize<'de> for Concurrency {
 pub trait ConcurrencyOption {
     fn parse_concurrency(&self, default: &Self) -> Option<usize>;
     fn is_none(&self) -> bool;
+    fn is_some(&self) -> bool {
+        !self.is_none()
+    }
```
Comment (bruceg) [inline src/sinks/util/service.rs:134]: Ah, I see, when I read `.is_none()` I forgot they weren't simple `Option`s.

File: src/sinks/util/service.rs [hunk 3]
```diff
@@ -180,8 +185,12 @@ pub struct TowerRequestConfig<T: ConcurrencyOption = Concurrency> {
 
 impl<T: ConcurrencyOption> TowerRequestConfig<T> {
     pub fn unwrap_with(&self, defaults: &Self) -> TowerRequestSettings {
+        // let concurrency
+
```
Comment (bruceg) [inline src/sinks/util/service.rs:189]: Is this a leftover from something?

**Review by bruceg — APPROVED** [review]

File: src/sinks/util/service.rs (hunk 3 above)
Comment (ktff) [reply src/sinks/util/service.rs]: Yep

**Pull request merged**